mech_client/marketplace_interact.py

```diff
@@ -586,51 +588,33 @@ def check_prepaid_balances(
     :type max_delivery_rate: int
     """
     requester = crypto.address
-    if payment_type == PAYMENT_TYPE_NATIVE:
-        with open(
-            Path(__file__).parent / "abis" / "BalanceTrackerFixedPriceNative.json",
-            encoding="utf-8",
-        ) as f:
-            abi = json.load(f)
-
-        balance_tracker_contract = get_contract(
-            contract_address=mech_payment_balance_tracker,
-            abi=abi,
-            ledger_api=ledger_api,
-        )
-        requester_balance = balance_tracker_contract.functions.mapRequesterBalances(
-            requester
-        ).call()
-        if requester_balance < max_delivery_rate:
-            print(
-                f"  - Sender Native deposited balance low. Needed: {max_delivery_rate}, Actual: {requester_balance}"
-            )
-            print(f"  - Sender Address: {requester}")
-            print("  - Please use scripts/deposit_native.py to add balance")
-            sys.exit(1)
 
-    if payment_type == PAYMENT_TYPE_TOKEN:
-        with open(
-            Path(__file__).parent / "abis" / "BalanceTrackerFixedPriceToken.json",
-            encoding="utf-8",
-        ) as f:
-            abi = json.load(f)
+    payment_type_name = (PaymentType(payment_type).name).lower()
```

Comment:
Is there any case for the `payment_type` to be unsupported? If yes, then we need to make this code safer as `PaymentType(payment_type)` would raise.

Reply:
Should not be an issue as I do the check inside the fetch mech info

Reply:
https://github.com/valory-xyz/mech-client/blob/feat/nvm-token/mech_client/marketplace_interact.py#L160

Reply:
ok, keep in mind that it would only be safe if the method accepted a `PaymentType`, which could be obtained after the check you linked because the code might be restructured in the future.